### PR TITLE
Implement `lt`, `lte`, `gt`, `gte`, `between` comparators.

### PR DESCRIPTION
**Pull request by BorysShulyak**
type: **minor**

## Changes
- Implement `lt`, `lte`, `gt`, `gte`, `between` comparators. Cover it with tests.
- Add tests to cover the `rules.some()` method usage. 

@mattmilburn Could you review the PR, please?

### CODE REVIEW

File: server/utils/__tests__/is-protected-entity.spec.js [hunk 1]
```diff
@@ -115,11 +115,107 @@ describe('isProtectedEntity', () => {
...
+    const rules = [
+      ['attribute', 'is', 0],
+      ['attribute', 'is', 2],
+    ];
+    const result = isProtectedEntity(entity, rules);
+
+    expect(result).toBe(false);
+  });
+
+  it('should return `true` if at least one of the provided rules is truly', () => {
```
Comment (mattmilburn) [inline server/utils/__tests__/is-protected-entity.spec.js]: @BorysShulyak Can you change the spelling from `truly` to `truthy`?

File: server/utils/__tests__/is-protected-entity.spec.js [hunk 2]
```diff
@@ -115,11 +115,107 @@ describe('isProtectedEntity', () => {
...
+    const entity = { attribute: 1 };
+    const rules = [
+      ['attribute', 'is', 0],
+      ['attribute', 'is', 1],
+    ];
+    const result = isProtectedEntity(entity, rules);
+
+    expect(result).toBe(false);
```
Comment (mattmilburn) [inline server/utils/__tests__/is-protected-entity.spec.js]: @BorysShulyak This should `expect` a `true` value, not `false`.

File: server/utils/__tests__/is-protected-entity.spec.js [hunk 3]
```diff
@@ -115,11 +115,107 @@ describe('isProtectedEntity', () => {
...
+    const entity = { attribute: 1 };
+    const rules = [['attribute', 'gt', 0]];
+    const result = isProtectedEntity(entity, rules);
+
+    expect(result).toBe(true);
+  });
+
+  it('should return `false` if entity attribute does not match the `gt` rule', () => {
+    const entity = { attribute: 1 };
+    expect(isProtectedEntity(entity, [['attribute', 'gt', 1]])).toBe(false);
```
Comment (mattmilburn) [inline server/utils/__tests__/is-protected-entity.spec.js:159]: @BorysShulyak Can you include a space between the `const` line and the `expect` lines?

**Comment by mattmilburn** [comment]
@BorysShulyak Looking great! I just posted a few minor comments. Once that is complete we can merge this and I'll get a new release ready for v1.1.0 of the plugin.

File: server/utils/__tests__/is-protected-entity.spec.js (hunk 1 above)
Comment (BorysShulyak) [reply server/utils/__tests__/is-protected-entity.spec.js]: Done.

File: server/utils/__tests__/is-protected-entity.spec.js (hunk 2 above)
Comment (BorysShulyak) [reply server/utils/__tests__/is-protected-entity.spec.js]: Done.

File: server/utils/__tests__/is-protected-entity.spec.js (hunk 3 above)
Comment (BorysShulyak) [reply server/utils/__tests__/is-protected-entity.spec.js:159]: Done.

**Comment by BorysShulyak** [comment]
@mattmilburn Applied your suggestions. Thank you!

**Comment by mattmilburn** [comment]
@BorysShulyak Excellent 🎉  Great work! Merging to staging now 🙂

**Pull request merged**